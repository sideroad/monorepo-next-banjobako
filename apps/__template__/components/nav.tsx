import React, { FC } from 'react';
import Link from 'components/link';
import { NAV } from '../cid';

const links = [{ href: '/users/octocat', label: 'Octocat' }];

const Nav: FC = () => (
  <nav data-cid={NAV}>
    <ul>
      <li>
        <Link href="/">
          <a>Home</a>
        </Link>
      </li>
      <ul>
        {links.map(({ href, label }) => (
          <li key={`nav-link-${href}-${label}`}>
            <Link href="/users/[id]" as={href}>
              <a>{label}</a>
            </Link>
          </li>
        ))}
      </ul>
    </ul>

    <style jsx>{`
      nav {
        text-align: center;
      }
      ul {
        display: flex;
        justify-content: space-between;
      }
      nav > ul {
        padding: 4px 16px;
      }
      li {
        display: flex;
        padding: 6px 8px;
      }
      a {
        color: #333;
        text-decoration: none;
        font-size: 13px;
      }
    `}</style>
  </nav>
);

export default Nav;
