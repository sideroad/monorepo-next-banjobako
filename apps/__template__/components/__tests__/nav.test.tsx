import React from 'react';
import Nav from '../nav';
import renderer from 'react-test-renderer';
import { init, Provider } from '@sideroad/react-i18n';

test('Nav to render navigation', () => {
  window.__SIMPLE_I18N__ = { resource: {}, lang: 'en' };

  const i18n = init({
    assignedLanguage: 'en'
  });
  const component = renderer.create(
    <Provider value={i18n}>
      <Nav />
    </Provider>
  );
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
