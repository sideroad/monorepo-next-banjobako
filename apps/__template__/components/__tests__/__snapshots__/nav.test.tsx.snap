// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Nav to render navigation 1`] = `
<nav
  className="jsx-144481089"
  data-cid="NwmE_uV-"
>
  <ul
    className="jsx-144481089"
  >
    <li
      className="jsx-144481089"
    >
      <a
        className="jsx-144481089"
        data-cid="SWTD9_x0"
        href="/en"
        onClick={[Function]}
        onMouseEnter={[Function]}
      >
        Home
      </a>
    </li>
    <ul
      className="jsx-144481089"
    >
      <li
        className="jsx-144481089"
      >
        <a
          className="jsx-144481089"
          data-cid="SWTD9_x0"
          href="/en/users/octocat"
          onClick={[Function]}
          onMouseEnter={[Function]}
        >
          Octocat
        </a>
      </li>
    </ul>
  </ul>
</nav>
`;
